syscalls.c

```diff
@@ -1183,12 +1390,21 @@ int syscalls_sys_getpeername(char *ustack)
 	GETFROMSTACK(ustack, struct sockaddr *, address, 1);
 	GETFROMSTACK(ustack, socklen_t *, address_len, 2);
 
+	if (vm_mapBelongs(proc, address, sizeof(*address)) < 0) {
```

Comment:
Size of `address` should be passed in `*address_len` (`address_len` must be not NULL). Maybe copy code from `syscalls_sys_getsockopt`.